plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -935,9 +967,29 @@ public DeltaLakeOutputTableHandle beginCreateTable(ConnectorSession session, Con
         for (ColumnMetadata column : tableMetadata.getColumns()) {
             columnNames.add(column.getName());
             columnNullabilities.put(column.getName(), column.isNullable());
-            columnTypes.put(column.getName(), serializeColumnType(NONE, new AtomicInteger(), column.getType()));
-            columnsMetadata.put(column.getName(), ImmutableMap.of());
-            columnHandles.add(toColumnHandle(column, column.getName(), column.getType(), partitionedBy));
+
+            Object serializedType = serializeColumnType(columnMappingMode, fieldId, column.getType());
+            Type physicalType = deserializeType(typeManager, serializedType, usePhysicalName);
```

Comment:
it looks weird to `serializeColumnType` only to `deserializeType` in the next line.
can this be simplified? 

also, why `serializeColumnType` doesn't need `usePhysicalName`?
the input, `column.getType()`, doesn't have physical and logical names, so how are `.../metadata/delta.columnMapping.physicalName` and `.../metadata/name` populated?

Reply:
> can this be simplified?

That's possible, but not so simple. Let me handle in follow-up. 

> also, why serializeColumnType doesn't need usePhysicalName?

The method uses `ColumnMappingMode` argument instead of the flag. I passed the mode because it affects not only `delta.columnMapping.physicalName` but also `delta.columnMapping.id`. 

> how are .../metadata/delta.columnMapping.physicalName and .../metadata/name populated?

* `delta.columnMapping.physicalName` is populated in serializeStructType → generateColumnMetadata in case of id and name mode
* `name` is always populated in serializeStructType → serializeStructField